tests/unit/test_time_step_control.py

```diff
@@ -8,85 +8,72 @@
 class TestParameterInputs:
     """The following tests are written to check the sanity of the input parameters"""
 
-    # TODO: Remove parametrization
-    # TODO: Purge dt_min_max from passed parameters
-    # TODO: Check for default values of dt_min_max
-    @pytest.mark.parametrize("constant_dt", [True, False])
-    def test_default_parameters_and_attribute_initialization(self, constant_dt):
+    def test_default_parameters_and_attribute_initialization(self):
         """Test the default parameters and initialization of attributes."""
-        tsc = Ts(
-            schedule=[0, 1], dt_init=0.2, constant_dt=constant_dt, dt_min_max=(0.1, 0.5)
-        )
+        tsc = Ts(schedule=[0, 1], dt_init=0.1)
         np.testing.assert_equal(tsc.schedule, np.array([0, 1]))
         assert tsc.time_init == 0
         assert tsc.time_final == 1
-        assert tsc.dt_init == 0.2
-        assert tsc.dt_min_max == (0.1, 0.5)
+        assert tsc.dt_init == 0.1
+        assert tsc.dt_min_max == (0.001, 0.1)
         assert tsc.iter_max == 15
         assert tsc.iter_optimal_range == (4, 7)
         assert tsc.iter_relax_factors == (0.7, 1.3)
         assert tsc.recomp_factor == 0.5
         assert tsc.recomp_max == 10
         assert tsc.time == 0
-        assert tsc.dt == 0.2
+        assert tsc.dt == 0.1
 
-    # def test_initialization_with_constant_dt(self):
-    #     """Test the proper initialization of parameters when setting a constant time step."""
-    #     tsc = Ts(schedule=[0,1], dt_init=1., constant_dt=True)
-
-    #     assert tsc.dt == 1
-    #     assert tsc.dt_min == 1
-    #     assert tsc.dt_max == 1
-    #     assert tsc.iter_low_factor == 1
-    #     assert tsc.iter_upp_factor == 1
-    #     assert tsc.recomp_factor == 1
-    #     assert tsc.is_constant
+    @pytest.mark.parametrize(
+        "schedule",
+        [(0, 0.5, 1), [0, 0.5, 1], np.array([0, 0.5, 1])]
+    )
+    def test_schedule_argument_type_compatibility(self, schedule):
```

Comment:
Great! Just for the sake of completeness. Could you add an assert that compares the given schedule (converted into an np.ndarray) with tsc.schedule?